### PR TITLE
fix: Barcode scanner rect - null if nothing found

### PR DESCRIPTION
**Pull request by bensonarafat**
#490 
Based on this issue
Rect can return null  
( https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode#public-rect-getboundingbox ) 

And also updated the example SDK, all the packages use `>=2.16.0 <3.0.0`

### CODE REVIEW

**Comment by fbernaly** [comment]
@bensonarafat : how did you test this? could you share a couple of barcode examples that come with null boundingBox? Did you test in iOS?

File: packages/example/lib/vision_detector_views/painters/barcode_detector_painter.dart [hunk 1]
```diff
@@ -43,21 +43,21 @@ class BarcodeDetectorPainter extends CustomPainter {
       builder.pop();
 
       final left = translateX(
-        barcode.boundingBox.left,
+        barcode.boundingBox!.left.toDouble(),
```
Comment (fbernaly) [inline packages/example/lib/vision_detector_views/painters/barcode_detector_painter.dart:46]: Avoid force unwrapping if you are uncertain or have not check if the property is not null

**Comment by bensonarafat** [comment]
> @bensonarafat : how did you test this? could you share a couple of barcode examples that come with null boundingBox? Did you test in iOS?

i did have an example I used that come up with null but what I noticed from the doc, that it might return null, that's why I shared a reference 
image

but what do you think, becoz i ain't perfect.

File: packages/example/lib/vision_detector_views/painters/barcode_detector_painter.dart (hunk 1 above)
Comment (bensonarafat) [reply packages/example/lib/vision_detector_views/painters/barcode_detector_painter.dart:46]: ok

**Comment by fbernaly** [comment]
@bensonarafat : could you share the image for the barcode that returns null for me to test with it.

**Pull request merged**